### ISSUE TITLE
Marian forgets best ce/bleu/perplexity after restarting from a checkpoint

### ISSUE THREAD

**Issue by afaji** [marian-nmt/marian-dev#180]
this impact early stopping criteria.

**Comment by emjotde** [comment]
We already working on that. Including saving Adam statistics. This should also save the state of the scheduler and current best validation results. 

Later this should also try to resume in roughly the same area in the training corpus. Probably only for the SQLite container.

**Comment by emjotde** [comment]
Issue #64

**Comment by snukky** [comment]
I store the following scheduler parameters:

* the best score from the first validation metric,
* the number of stalled validation steps,
* the number of epochs and batches,
* the actual learning rate and multiplication factor.

Do we need something else?

**Comment by emjotde** [comment]
How about the direction of validation, i.e. increasing/decreasing scores are better?
